samples/nrf_desktop/src/modules/config.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdlib.h>
+
+#include <zephyr.h>
+#include <settings/settings.h>
+
+#include "event_manager.h"
+#include "config_event.h"
+
+#define MODULE config
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CONFIG_LOG_LEVEL);
+
+#define KEY_LEN 9
+
+/* Array of pointers to data loaded from settings */
+static u8_t *loaded_data[CONFIG_EVENT_ID_MAX];
+
+static int settings_set(int argc, char **argv, void *val_ctx)
+{
+	if (argc != 1) {
+		return -ENOENT;
+	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_MOTION_OPTICAL_ENABLE)) {
+		char *end;
+		u8_t id = strtol(argv[0], &end, 10);
+		if (*end != '\0') {
+			LOG_WRN("Loaded config ID is not a valid number");
+			return 0;
+		}
+
+		if ((id == CONFIG_EVENT_ID_RESERVED) ||
+		    (id >= CONFIG_EVENT_ID_MAX)) {
+			LOG_WRN("Not supported config: %u", id);
+			return 0;
+		}
+
+		__ASSERT_NO_MSG(id > CONFIG_EVENT_ID_RESERVED);
```

Comment:
there is no point on asserting this - it was checked few lines above

Reply:
done